superbit_lensing/medsmaker/superbit/medsmaker_mocks.py

```diff
@@ -411,97 +454,208 @@ def make_psf_models(self, select_stars=False, use_coadd=False):
-        """
+
         psfex_model_file=imcat_ldac_name.replace('.ldac','.psf')
+
         # Just return name, the make_psf_models method reads it in as a PSFEx object
         return psfex_model_file
 
-    def _select_stars_for_psf(self,sscat,truthfile):
+    def _make_piff_model(self, imagefile, weightfile='weight.fits', sextractor_config_path=None,
+            psfex_out_dir='./tmp/', select_truth_stars=False,star_params=None):
+        '''
+        Method to invoke PIFF for PSF modeling
+        Also creates individual exposure catalogs
+        Unclear what it should return yet...
+        Steps:
+            - Create sextractor catalog for input image
+            - Filter sexcat to create star catalog on the fly
+              with some common-sense selections
+            - If option is given, match star cat against truth
+            - Run PIFF
+            -
+
+        First, let's get it to run on one, then we can focus on running list!
+        '''
+
+        if sextractor_config_path is None:
+                sextractor_config_path = os.path.join(self.base_dir, 'superbit/astro_config/')
+
+        imcat_ldac_name=imagefile.replace('.fits','_cat.ldac')
+
+        if select_truth_stars==True:
+            # This will break for any truth file nomenclature
+            # that isn't pipeline default
+            import glob
+            truthdir=self.data_dir
+            try:
+                truthcat = glob.glob(os.path.join(truthdir,'truth*.dat'))[0]
+            except:
+                truthcat = glob.glob(os.path.join(truthdir,'truth*.fits'))[0]
+            truthfilen=os.path.join(truthdir,truthcat)
+            self.logprint("using truth catalog %s" % truthfilen)
+            psfcat_name = self._select_stars_for_psf(sscat=imcat_ldac_name,\
+                    truthfile=truthfilen,star_params=star_params)
+
+        else:
+            psfcat_name = self._select_stars_for_psf(sscat=imcat_ldac_name,\
+                    truthfile=None,star_params=star_params)
+
+        # Now run PIFF on that image and accompanying catalog
+        piff_config_arg = ''.join([sextractor_config_path,'piff.config'])
+        im_name = imagefile.replace('.fits','.sub.fits')
+        image_arg = ''.join(['input.image_file_name=',im_name])
+        psfcat_arg = ''.join(['input.cat_file_name=',psfcat_name])
+        output_name = imagefile.split('/')[-1].replace('.fits','.piff')
+        output_dir = os.path.join(self.data_dir,'piff-output')
+        full_output_name=os.path.join(output_dir,output_name)
+        output_arg = ''.join(['output.file_name=',output_name,' output.dir=',output_dir])
+        cmd = ' '.join(['piffify', piff_config_arg,image_arg,psfcat_arg, output_arg])
+
+        self.logprint("piff cmd is " + cmd)
+        os.system(cmd)
+
+        piff_extended = self.piff_extender(full_output_name)
+
+        return piff_extended
+
+    def piff_extender(self,piff_file,stamp_size=21):
```

Comment:
I think it makes more sense to define `piff_extender()` outside of the `BITMeasurement` class - it's more of a helper function, as it doesn't depend on `self` at all.

Reply:
Is that better convention?

Reply:
Ya. In general it is better to put things within their scope. `piff_extender()` is not an instance method of `BITMeasurement` because it doesn't use any of the properties of the instance. Now this could be rewritten so that `full_output_name` isn't passed to the method and it gets this from a `self.full_output_name` call, but the PIFF extender is useful in general and we might want to use it outside the context of this class

Reply:
(to be clear - as it is written it currently _is_ an instance method as it explicitly has `self` in the function definition, but it doesn't _need_ to be as `self` is never used. That is what I mean. The other option is to put the `@staticmethod` decorator above the function def, but I prefer it to be outside of the class for the reason given above)